src/app/Mail/RabbitMqConnection.php

```diff
@@ -0,0 +1,308 @@
+                // Filter the message by topic
+                $status = $this->channel->basic_publish($msg, $exchange, $routingKey);
+            }
+            $this->channel->wait_for_pending_acks();
+            // $this->channel->wait_for_pending_acks_returns();
+
+            return $status;
+        } catch (\Throwable $e) {
+            // the above code throws various kinds of exceptions-- convert them
+            // all to an Exception
+            throw self::convertException($e);
+        }
+    }
+
+    /**
+     * return default AMQP message headers as an array
+     *
+     * @return array an array containing default AMQP message headers
+     */
+    public function getDefaultHeaders(): array
+    {
+        return $this->defaultHeaders;
+    }
+
+    /**
+     * return client config
+     * @codeCoverageIgnore
+     * @throws Exception
+     */
+    protected function initConfig(string $configName, array $config = [], array $headers = []): array
+    {
+        if (empty($config)) {
+            $config = Config::load($configName);
+            $config = $config->toArray();
+        }
+
+        if (!empty($headers)) {
+            $this->setDefaultHeaders($headers);
+        }
+
+        return $config;
+    }
+
+    /**
+     * set the default AMQP messsage headers; these can be amended later when calling send()
+     *
+     * @return array returns the newly set headers as an array
+     */
+    public function setDefaultHeaders(array $headers): array
+    {
+        $this->defaultHeaders = $headers;
+        return $this->defaultHeaders;
+    }
+
+    /**
+     * set the default AMQP messsage properties
+     *
+     * @return array an array containing the message properties
+     */
+    protected function getMessageProps(): array
+    {
+        return [
+            'content_type'  => 'application/json',
+            // 'delivery_mode' => AMQPMessage::DELIVERY_MODE_PERSISTENT
+        ];
+    }
+
+    /**
+     * create an AMQP message object with the specified data, properties, and headers
+     *
+     * @param string|array $data a string or array containing the message payload; arrays will be automatically converted to JSON format
+     * @param array $props an array containing the message properties
+     * @param array $headers an array containing the message headers
+     * @return AMQPMessage an AMQPMessage object
+     */
+    protected function createAMQPMessage($data, array $props, array $headers): AMQPMessage
+    {
+        if (is_array($data)) {
+            $data = json_encode($data);
+        }
+
+        $msg = new AMQPMessage($data, $props);
+        $msg->set('application_headers', new AMQPTable($headers));
+        return $msg;
+    }
+
+    /**
+     * clean up an AMPQ connection
+     *
+     * @returns void
+     */
+    protected static function cleanupConnection($connection)
+    {
+        // cleanup the connection and mask any exceptions
+        try {
+            if ($connection !== null) {
```

Comment:
It will be nice if you change to
if (!$connection) { 
    return; 
}

try { ... } catch(...)